presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java

```diff
@@ -0,0 +1,170 @@
+    {
+        transactionManager.tryRegisterTransaction(transactionInfo);
+        List<QualifiedObjectName> viewsList = metadataProvider.listViews(session.toSession(sessionPropertyManager), prefix);
+        if (!viewsList.isEmpty()) {
+            writeValueAsString(viewsList, objectMapper);
+        }
+        return EMPTY_STRING;
+    }
+
+    @ThriftMethod
+    public String getViews(TransactionInfo transactionInfo, SessionRepresentation session, QualifiedTablePrefix prefix)
+    {
+        transactionManager.tryRegisterTransaction(transactionInfo);
+        Map<QualifiedObjectName, ViewDefinition> viewsMap = metadataProvider.getViews(session.toSession(sessionPropertyManager), prefix);
+        if (!viewsMap.isEmpty()) {
+            return writeValueAsString(viewsMap, objectMapper);
+        }
+        return EMPTY_STRING;
+    }
+
+    @ThriftMethod
+    public String getView(TransactionInfo transactionInfo, SessionRepresentation session, QualifiedObjectName viewName)
+    {
+        transactionManager.tryRegisterTransaction(transactionInfo);
+        Optional<ViewDefinition> viewDefinition = metadataProvider.getView(session.toSession(sessionPropertyManager), viewName);
+        if (viewDefinition.isPresent()) {
+            return writeValueAsString(viewDefinition.get(), objectMapper);
+        }
+        return EMPTY_STRING;
+    }
+
+    @ThriftMethod
+    public String getMaterializedView(TransactionInfo transactionInfo, SessionRepresentation session, QualifiedObjectName viewName)
+    {
+        transactionManager.tryRegisterTransaction(transactionInfo);
+        Optional<ConnectorMaterializedViewDefinition> connectorMaterializedViewDefinition = metadataProvider.getMaterializedView(session.toSession(sessionPropertyManager), viewName);
+        if (connectorMaterializedViewDefinition.isPresent()) {
+            return writeValueAsString(connectorMaterializedViewDefinition.get(), objectMapper);
+        }
+        return EMPTY_STRING;
+    }
+
+    @ThriftMethod
+    public String getReferencedMaterializedViews(TransactionInfo transactionInfo, SessionRepresentation session, QualifiedObjectName tableName)
+    {
+        transactionManager.tryRegisterTransaction(transactionInfo);
+        List<QualifiedObjectName> referencedMaterializedViewsList = metadataProvider.getReferencedMaterializedViews(session.toSession(sessionPropertyManager), tableName);
+        if (!referencedMaterializedViewsList.isEmpty()) {
+            return writeValueAsString(referencedMaterializedViewsList, objectMapper);
+        }
+        return EMPTY_STRING;
+    }
+
+    private static String writeValueAsString(Object value, ObjectMapper objectMapper)
+    {
+        try {
+            return objectMapper.writeValueAsString(value);
+        }
+        catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+}
```

Comment:
Would that be possible in this case?

Reply:
From what I know I'm not sure if that could be done as it wouldn't know the interface implementation type when deserializing. Though if there is a way to serialize and deserialize the metadata objects as a thrift struct object that would probably be better used here

Reply:
Json in the short term is fine but have a plan to get everything on Thrift.
Regarding
> if that could be done as it wouldn't know the interface implementation type when deserializing
Take a look at [Any](https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/server/thrift/Any.java) which I introduced to solve this exact problem.

Reply:
Sure, I will take a look at that thanks. If there can be a solution with it I will change to that, though it might be in a separate PR due to to others I have being blocked by this